lib/wos_records.rb

```diff
@@ -27,6 +27,22 @@ def initialize(records: nil, encoded_records: nil)
     @doc = Nokogiri::XML(@records) { |config| config.strict.noblanks }
   end
 
+  # Group records by the database prefix in the UID
+  #  - where a database prefix is missing, groups records into 'MISSING_DB'
+  # @return [Hash<String => WosRecords>]
+  def by_database
+    nodes = doc.dup.search('REC')
```

Comment:
I'm fairly certain that this `.dup` is no longer required, since every value in the Hash returned is a new `WosRecords` object based on the serialized representation (of already serialized elements).  Nothing attached to the old doc is returned.  Can you construct a test related to the question of immutability that would fail without the dup?  I might be missing something, but I can't imagine one right now.

Side note: `WosRecords` might be more useful if it could (also) be initialized on the Array of records (here `db_recs[db]`).

Reply:
Pushed up an amended commit to remove the `doc.dup`; leaving the init change to another PR (could be good if we find additional uses for it outside this particular method).